app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,23 @@
+class RelationshipsController < ApplicationController
+  before_action :require_login
+
+  def index
+    @following_relationships = current_user.following_relationships
+  end
+
+  def create
+    user = User.find(params[:leader_id])
+    if current_user.can_follow?(user)
+      Relationship.create(follower: current_user, leader: user)
```

Comment:
Anytime you don't check if a creation succeeds via an `if`/`else`, it is best to use the bang version (`create!` for example) so that an exception will be raised if there is a problem. Otherwise, bad data can sneak in your database and it can be hard to debug where it came from.